src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/DepositExtractor.cs

```diff
@@ -27,7 +27,10 @@ public class DepositExtractor : IDepositExtractor
             IFullNode fullNode)
         {
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
-            this.depositScript = federationGatewaySettings.MultiSigRedeemScript;
+            // Note: MultiSigRedeemScript.PaymentScript equals MultiSigAddress.ScriptPubKey
+            this.depositScript =
+                federationGatewaySettings?.MultiSigRedeemScript?.PaymentScript ??
+                federationGatewaySettings?.MultiSigAddress?.ScriptPubKey;
```

Comment:
If `MultiSigRedeemScript.PaymentScript` equals `MultiSigAddress.ScriptPubKey` can't we consolidate the properties ? I am confused as to why you would select `ScriptPubKey` when `PaymentScript` is `null`.